pkg/operator/staticpod/controller/startupmonitorcondition/startup_monitor_condition.go

```diff
@@ -89,16 +95,24 @@ func (fd *startupMonitorPodConditionController) sync(ctx context.Context, _ fact
 
 	if len(currentNodeName) == 0 {
 		// nothing to do, no node is progressing
+		startupPodDegraded = startupPodDegraded.WithStatus(operatorv1.ConditionFalse)
+		startupPodContainerExcessiveRestartsDegraded = startupPodContainerExcessiveRestartsDegraded.WithStatus(operatorv1.ConditionFalse)
 		return nil
 	}
 
 	monitorPod, err := fd.podLister.Get(fmt.Sprintf("%s-startup-monitor-%s", fd.targetName, currentNodeName))
 	if errors.IsNotFound(err) {
+		startupPodDegraded = startupPodDegraded.WithStatus(operatorv1.ConditionFalse)
+		startupPodContainerExcessiveRestartsDegraded = startupPodContainerExcessiveRestartsDegraded.WithStatus(operatorv1.ConditionFalse)
 		return nil
 	} else if err != nil {
 		return err // unknown error
 	}
 
+	// by default, both conditions are in a non-degraded state
+	startupPodDegraded = startupPodDegraded.WithStatus(operatorv1.ConditionFalse)
+	startupPodContainerExcessiveRestartsDegraded = startupPodContainerExcessiveRestartsDegraded.WithStatus(operatorv1.ConditionFalse)
+
```

Comment:
nit: Personally, I’d restructure this controller so that every code path sets status explicitly instead of relying on a default value. I think it would make things simpler to have a consistent approach in this controller. What do you think?

Reply:
To be clear: I don't think this is a big deal and I'm fine with merging as is if you disagree

Reply:
I would prefer to set the default values here because `startupPodDegraded` can be set to `true` in `withDegradedWhenInPendingPhaseFor` and then we would have to add checks not to revert it to `false` in `withDegradedWhenHasFailedContainer`. Does it make sense ?